src/client/workflow/workflow.ts

```diff
@@ -0,0 +1,262 @@
+import { nanoid } from "nanoid";
+import type { AsyncStepFunction, PARALLEL_CALL_STATE, Step, StepFunction } from "./types";
+import { internalHeader, workflowIdHeader } from "./types";
+import type { Client } from "../client";
+
+export class Workflow {
+  protected readonly client: Client;
```

Comment:
Instead of 
```typescript
const workflow = new Workflow({client: new Client(), ...})
```
We can do 
```typescript
const workflow = new Client().workflow()
```

to preserve the convention.

Reply:
I didn't intend the current version to be how the user creates a workflow. We discussed having a Serve method to use workflows with something like:

```ts
// route.ts
import { serve } from "@upstash/qstash"

client = // ...

const POST = serve(async (context) => {
  const paylaod = context.requestPayload() // return initial request payload
  const result1 = context.run("step1", async () => {...})
})
```